styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java

```diff
@@ -26,22 +26,23 @@
 import com.cronutils.utils.VisibleForTesting;
 import com.google.common.base.Preconditions;
 import com.spotify.styx.model.ExecutionDescription;
+import com.spotify.styx.model.SequenceEvent;
 import com.spotify.styx.model.WorkflowInstance;
 import com.spotify.styx.monitoring.Stats;
 import com.spotify.styx.publisher.Publisher;
-import com.spotify.styx.state.OutputHandler;
 import com.spotify.styx.state.RunState;
 import com.spotify.styx.util.Retrier;
 import com.spotify.styx.util.RunnableWithException;
 import java.time.Duration;
 import java.util.Objects;
+import java.util.function.BiConsumer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * An {@link OutputHandler} that integrates {@link RunState.State} values with a {@link Publisher}.
+ * An event consumer that integrates {@link RunState.State} values with a {@link Publisher}.
  */
-public class PublisherHandler implements OutputHandler {
+public class PublisherHandler implements BiConsumer<SequenceEvent, RunState> {
```

Comment:
Not as simple as I thought due to the fanout thing.

Reply:
Tried in https://github.com/spotify/styx/pull/691

Reply:
I guess it could be rewritten to act on the event instead of switching on the new state. But that's just a cosmetic change.

Another approach here could be to still have `PublisherHandler` be an `OutputHandler` so it gets called at-least-once, and e.g. keep "internal" (possibly) persisted state or specially craft pubsub messages to be idempotent and/or effectively publish exactly-once. I considered looking into that but I do not believe the at-most-once behavior of PublisherHandler to be a regression (it matches current behavior before this PR) and so can be considered out of scope for this PR. I'd rather revisit in another PR if necessary.

Reply:
Here's a PR that make the `PublisherHandler` observe the event instead of the state. It feels a bit pointless though. https://github.com/spotify/styx/pull/692